coffee_core/src/coffee.rs

```diff
@@ -272,11 +273,31 @@ impl PluginManager for CoffeeManager {
         })
     }
 
-    async fn upgrade(&mut self, _: &[&str]) -> Result<(), CoffeeError> {
+    async fn upgrade(&mut self, repo: &str, all: bool) -> Result<CoffeeUpgrade, CoffeeError> {
         self.remote_sync().await?;
-        // FIXME: Fix debug message with the list of plugins to be upgraded
-        log::debug!("upgrading plugins");
-        Ok(())
+        let mut total_status: Vec<CoffeeRepositoryUpgrade> = vec![];
+        if all {
+            for repo in self.repos.values() {
+                let result = repo.upgrade(&self.config.plugins).await?;
+                total_status.push(result);
+            }
+        } else {
+            let repository = self
+                .repos
+                .get(repo)
+                .ok_or(error!("repository with name: {repo} not found"))?;
+            let status = repository.upgrade(&self.config.plugins).await?;
```

Comment:
Looks like that there is a log somewhere

By running the following command

```
➜  coffee git:(pr132-review) ✗ cargo run --bin coffee -- --data-dir /home/vincent/clightning4j --network testnet upgrade lightningd
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/coffee --data-dir /home/vincent/clightning4j --network testnet upgrade lightningd`
[2023-05-18T02:20:13Z ERROR coffee_lib::errors] ERROR #1: repository with name: lightningd not found
Remote repository lightningd is already up to date!
➜  coffee git:(pr132-review) ✗ 

```

Why we have the `ERROR coffee_lib::errors] ERROR #1:`?

Reply:
I used `ok_or_else` instead of `ok_or` to fix this
It is invoked lazily, meaning the error message expression is only evaluated if the Option is None.